pwiz_tools/Skyline/Controls/Graphs/AreaReplicateGraphPane.cs

```diff
@@ -1,1473 +1,1518 @@
-                                              ZOrder = ZOrder.F_BehindGrid,
-                                          };
-
-
-                    textObj.FontSpec.Border.IsVisible = false;
-                    textObj.FontSpec.Size = pointSize.Value;
-                    _labelHeight =(int) textObj.FontSpec.GetHeight(CalcScaleFactor());
-                    GraphObjList.Add(textObj);
-                    _dotpLabels.Add(textObj);
-                }
-            }
-        }
-
-        private int? GetDotProductsPointSize(Graphics g)
-        {
-            for (int pointSize = (int) Settings.Default.AreaFontSize; pointSize > 4; pointSize--)
-            {
-                var fontLabel = new FontSpec {Size = pointSize};
-                var sizeLabel = fontLabel.MeasureString(g, DotpLabelText, CalcScaleFactor());
-
-                float labelWidth = (float) XAxis.Scale.ReverseTransform((XAxis.Scale.Transform(0) + sizeLabel.Width));
-
-                if (labelWidth < 1.2)
-                    return pointSize;          
-            }
-            return null;
-        }
-
-        private string GetDotProductResultsText(int indexResult)
-        {
+                _dotpData = ImmutableList.ValueOf(graphData.DotpData.Select(point => (float)point.Y));
+
+            if (ExpectedVisible != AreaExpectedValue.none &&
+                DotProductDisplayOption.line.IsSet(Settings.Default))
+            {
+                Y2Axis.IsVisible = true;
+                Y2Axis.Scale.Min = 0;
+                Y2Axis.Scale.Max = 1.1;
+                Y2Axis.Title.Text = DotpLabelText;
+                var dotpLine = new LineItem(DotpLabelText, graphData.DotpData, Color.DimGray, SymbolType.Circle )
+                {
+                    IsY2Axis = true, Line = new Line() { Style = DashStyle.Dash, Color = Color.DimGray, Width = 2.0f},
+                    Symbol = new Symbol() { Type = SymbolType.Diamond, Size = 5f, Fill = new Fill(Color.DimGray)}
+                };
+                CurveList.Insert(0, dotpLine);
+                ToolTip.TargetCurves.ClearAndAdd(dotpLine);
+            }
+            else
+            {
+                Y2Axis.IsVisible = false;
+            }
+
+            UpdateAxes(resetAxes, aggregateOp, dataScalingOption, normalizeOption);
+
+            if (Settings.Default.PeakAreaDotpCutoffShow && DotProductDisplayOption.line.IsSet(Settings.Default) && _dotpData != null)
+            {
+                var cutoff = Settings.Default.PeakAreaDotpCutoffValue;
+                var highlightValues = new PointPairList(graphData.DotpData.Select(point => point.Y < cutoff ? point : new PointPair(){X = point.X, Y = float.NaN}).ToList());
+                var cutoffHighlightLine = new LineItem("", highlightValues, Color.DimGray, SymbolType.Circle)
+                {
+                    IsY2Axis = true,
+                    Line = new Line() { Color = Color.Transparent},

```

Comment:
What is the effect of this line of code? Is it overwriting the "Color.DimGray" that was specified three lines above this?

Reply:
In my experience the color specified in the constructor arguments is ignored, but that argument is required, so I had to redefine it in the attribute definitions.